libs/application/templates/inheritance-report/src/forms/debts.ts

```diff
@@ -0,0 +1,218 @@
+                    id: 'balance',
+                    currency: true,
+                  },
+                ],
+                repeaterButtonText: m.debtsRepeaterButton.defaultMessage,
+                repeaterHeaderText: m.debtsCreditorHeader.defaultMessage,
+                sumField: 'balance',
+              },
+            ),
+          ],
+        }),
+      ],
+    }),
+    buildSubSection({
+      id: 'publicCharges',
+      title: m.publicChargesTitle,
+      children: [
+        buildMultiField({
+          id: 'publicCharges',
+          title: m.publicChargesTitle,
+          description: '',
+          children: [
+            buildDescriptionField({
+              id: 'publicChargesHeader',
+              title: m.publicChargesTitle,
+              description: m.publicChargesDescription,
+              titleVariant: 'h3',
+            }),
+            buildDescriptionField({
+              id: 'publicCharges.total',
+              title: '',
+            }),
+            buildCustomField(
+              {
+                title: '',
+                id: 'publicCharges.data',
+                component: 'ReportFieldsRepeater',
+              },
+              {
+                fields: [
+                  {
+                    title: m.amount.defaultMessage,
+                    id: 'publicChargesAmount',
+                    width: 'full',
+                    currency: true,
+                  },
+                ],
+                repeaterButtonText:
+                  m.publicChargesRepeaterButton.defaultMessage,
+                repeaterHeaderText: m.publicChargeHeader.defaultMessage,
+                sumField: 'publicChargesAmount',
+              },
+            ),
+          ],
+        }),
+      ],
+    }),
+    buildSubSection({
+      id: 'debtsOverview',
+      title: m.overview,
+      children: [
+        buildMultiField({
+          id: 'debtsOverview',
+          title: m.overview,
+          description: '',
+          children: [
+            buildDividerField({}),
+            buildDescriptionField({
+              id: 'overviewFuneralCost',
+              title: m.funeralCostTitle,
+              titleVariant: 'h3',
+              marginBottom: 'gutter',
+              space: 'gutter',
+            }),
+            buildKeyValueField({
+              label: m.totalAmount,
+              value: ({ answers }) =>
+                formatCurrency(String(answers.funeralCostAmount)),
+            }),
+            buildDividerField({}),
+            buildDescriptionField({
+              id: 'overviewDomesticAndForeignDebts',
+              title: m.debtsTitle,
+              titleVariant: 'h3',
+              marginBottom: 'gutter',
+              space: 'gutter',
+            }),
+            buildKeyValueField({
+              label: m.totalAmount,
+              value: ({ answers }) =>
+                formatCurrency(
+                  String((answers.domesticAndForeignDebts as any)?.total),
```

Comment:
I think you can import like this
```typescript
import { infer as zinfer } from 'zod'
import { inheritanceReportSchema } from '../../lib/dataSchema'
type InheritanceSchema = zinfer<typeof inheritanceReportSchema>

```

and then do something slightly awful, but typed, like
```typescript
(answers as unknown as InheritanceSchema).domesticAndForeignDebts?.total
```

But this relies on a dataschema being there for it.
Just food for thought.

Reply:
Deffs. I will need to update quite a few values when schema (and types) are somewhat established